File: backend/apps/owasp/graphql/nodes/sponsor.py
"""OWASP sponsors GraphQL node."""

import strawberry_django

from apps.owasp.models.sponsor import Sponsor


@strawberry_django.type(
    Sponsor,
    fields=[
        "image_url",
        "name",
        "sponsor_type",
        "url",
    ],
)
class SponsorNode:
    """Sponsor node."""

File: frontend/src/types/sponsor.ts
export type SponsorTier = 'Diamond' | 'Platinum' | 'Gold' | 'Silver' | 'Supporter'

export type SponsorData = {
  id: string
  imageUrl: string
  name: string
  sponsorType: string
  url: string
  /** Diamond tier: short line when API provides it (GraphQL may add later). */
  motto?: string
}

export type SponsorsByTier = {
  tier: SponsorTier
  sponsors: SponsorData[]
}

export const TIER_ORDER: SponsorTier[] = ['Diamond', 'Platinum', 'Gold', 'Silver', 'Supporter']
